langchain/src/callbacks/manager.ts

```diff
@@ -309,6 +311,47 @@ export class CallbackManager
     );
   }
 
+  async handleChatModelStart(
+    llm: { name: string },
+    messages: BaseChatMessage[][],
+    runId: string = uuidv4()
+  ): Promise<CallbackManagerForLLMRun> {
+    let messageStrings: string[];
+    await Promise.all(
+      this.handlers.map(async (handler) => {
+        if (!handler.ignoreLLM) {
+          try {
+            if (handler.handleChatModelStart)
+              await handler.handleChatModelStart?.(
```

Comment:
I'm sure we could make this more concise with some ternary ops or better use of conditional chaining...